src/compiler/sys.ts

```diff
@@ -494,9 +511,16 @@ namespace ts {
                     }
                 },
                 createHash(data) {

```

Comment:
consider defining two functions instead:
```ts
function createMD5HashUsingNativeCrypto(data: string) {
    const hash = _crypto.createHash("md5");
    hash.update(data);
    return hash.digest("hex");
}
function createDjb2Hash(data) {
    const chars = data.split("").map(str => str.charCodeAt(0));
    return `${chars.reduce((prev, curr) => ((prev << 5) + prev) + curr, 5381)}`;
}
```
then in here make createHash either one or the other based on the existance of _crypto,e.g.:

```ts
{
...
createHash: _crypto ? createMD5HashUsingNativeCrypto : createDjb2Hash
...
}
```